src/config.rs

```diff
@@ -12,12 +13,24 @@ pub(crate) struct Config {
 }
 
 impl Config {
-    pub fn load() -> Result<Self, Error> {
-        let Ok(file) = File::open("config.json") else {
+    pub fn load(default_config_path: Option<PathBuf>) -> Result<Self, Error> {
+        let override_config_path = PathBuf::from("config.json");
+
+        let final_path = if override_config_path.is_file() {
+            override_config_path
+        } else if let Some(path) = default_config_path {
+            if !path.is_file() {
+                return Err(Error::NoConfigFileFound);
+            }
+            path
+        } else {
+            return Err(Error::NoConfigFileFound);
+        };
+
+        let Ok(file) = File::open(final_path) else {
             return Err(Error::FileIoError);
         };
```

Comment:
```suggestion
```suggestion
        let config_path = {
            let override_config_path = PathBuf::from("config.json");

            if override_config_path.is_file() {
                Some(override_config_path)
            } else {
                None
            }
        }
        .or(default_config_path)
        .ok_or(Error::NoConfigFileFound)?;

        let file = File::open(config_path).or(Err(Error::FileIoError))?;
```